client/src/intro-exit/Surveys/quality_control.jsx

```diff
@@ -1,26 +1,39 @@
-import React, { useEffect } from "react";
+import React, { useEffect, useState } from "react";
 import SurveyWrapper from "../../components/SurveyWrapper";
 import surveyJSON from './quality_control.json';
 import { usePlayer, useGame } from "@empirica/player";
+import { useCallback } from "react";
 
 
 
 export default function quality_control({ next }) {
     const player = usePlayer();
     const game = useGame();
-    
+    const [displayPayment, setDisplayPayment] = useState("calculating...");
     useEffect(() => {
+        console.log("isPaidTime before" + player.get("isPaidTime"))
         player.set("isPaidTime", false); //stop paying participant when they get to this screen (so we can compute the time)
+        console.log("isPaidTime after" + player.get("isPaidTime"))
         console.log("QC Exit. Played for " + player.get("activeMinutes") + " minutes, earned $" + player.get("dollarsOwed"))
     }, [])
+
+    
     const dollarsOwed = player.get("dollarsOwed");
+    useEffect(() => {
+        setDisplayPayment(dollarsOwed)
+    }, [dollarsOwed])
 
+    useEffect(() => {
+        player.set("stopPaying", true)
```

Comment:
Werent we going to set `stopPaying` in the previous useEffect, with the setDisplayPayment? That way it didn't run before the payment calculation callback had finished? The way it is here, wouldn't it get called as soon as the component loads the first time, and possibly happen before the payment calculation callback happened?